docs: introduce docs on how to deploy to the platform, and help page for the licensing tool (#2479)
TimefoldAI/timefold-solver: docs/src/modules/ROOT/pages/deploying-to-platform/model-metadata.adoc

[#_platform_model_metadata]
= Platform model metadata
:description: Platform-specific model metadata that shapes the score analysis, configuration profile, and model identity UI.
:doctype: book
:sectnums:
:icons: font

Once your model is xref:deploying-to-platform/getting-started.adoc[deployed to Timefold Platform], the platform UI surfaces several pieces of your model's code and configuration directly.

This page documents that platform-specific metadata: which identity properties describe your model on the platform, how to give your constraints human-readable names and groups, and how to ship a default configuration profile.

> See xref:deploying-to-platform/introduction.adoc[Deploying to the Timefold Platform] for why you'd want to do this in the first place, and xref:deploying-to-platform/introduction.adoc#_platform_concepts[Platform concepts] for how this metadata relates to your model's identifier, registration key, and visibility on the platform.

Reviewer: ```suggestion
See xref:deploying-to-platform/introduction.adoc[Deploying to the Timefold Platform] for why you'd want to do this, and xref:deploying-to-platform/introduction.adoc#_platform_concepts[Platform concepts] for how this metadata relates to your model's identifier, registration key, and visibility on the platform.
```

            "Conflicts",
            "Constraints that prevent double-booking of teachers and rooms.",
            "alert-triangle",
            arrayOf("hard-constraints")
        )
    }

    fun roomConflict(constraintFactory: ConstraintFactory): Constraint {
        return constraintFactory
                // constraint implementation excluded
                .asConstraint(ConstraintInfo(ROOM_CONFLICT, "Room conflict",
                        "A room can be used for at most one lesson at the same time.", CONFLICT_GROUP))
    }

    fun teacherConflict(constraintFactory: ConstraintFactory): Constraint {
        return constraintFactory
                // constraint implementation excluded
                .asConstraint(ConstraintInfo(TEACHER_CONFLICT, "Teacher conflict",
                        "A teacher can teach at most one lesson at the same time.", CONFLICT_GROUP))
    }

    // other constraints excluded
}
----
--
====

The `id` you pass into `ConstraintInfo` is still the identifier used by `@ConstraintReference` in your `ModelConfigOverrides`, as described in xref:running-timefold-solver/service/constraint-overrides.adoc[Adjusting constraint weights].

> The `name`, `description`, and `constraintGroup` are purely presentational: they are what the platform's score analysis view and configuration profile editor use to show your constraints to consumers, instead of falling back to the raw id.

Reviewer: ```suggestion
The `name`, `description`, and `constraintGroup` are purely presentational: they are what the platform's score analysis view and configuration profile editor use to show your constraints to consumers, instead of falling back to the raw ID.
```
As it's not a quoted string `id`.

This profile is what consumers get out of the box, before they add any profile of their own.

[cols="2,3", options="header"]
|===
| Property | Purpose

| `ai.timefold.model.default-config.name`
| Name of the default configuration profile.

| `ai.timefold.model.default-config.description`
| Description of the default configuration profile.

| `ai.timefold.model.default-config.max-thread-count`
| Default number of threads used for solving.

| `ai.timefold.model.default-config.map.provider`
| Default map provider, for models that require map data.

| `ai.timefold.model.default-config.map.location`
| Default map location, for models that require map data.

| `ai.timefold.model.default-config.map.max-distance-from-road`
| Default maximum distance from a road, for models that require map data.

| `ai.timefold.model.default-config.map.transport-type`
| Default transport type used when computing distances, for models that require map data.

| `ai.timefold.model.default-config.map.use-traffic`
| Whether the default profile takes live traffic into account. Defaults to `false`.

| `ai.timefold.model.default-config.termination.spent-limit`

> | The default maximum solving time, in the https://www.digi.com/resources/documentation/digidocs/90001488-13/reference/r_iso_8601_duration_format.htm[ISO 8601 duration format]. This property is *required*.

Reviewer 2: Could we refer to Wikipedia instead of this Digi website? https://en.wikipedia.org/wiki/ISO_8601#Durations
```suggestion
| The default maximum solving time, in the https://en.wikipedia.org/wiki/ISO_8601#Durations[ISO 8601 duration format]. This property is *required*.
```
